instagram/src/App.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+
+import Authenticate from './components/Authentication/Authenticate';
+import PostsPage from './components/PostsPage/PostsPage'
+import dummyData from './dummy-data';
+
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      filter: '',
+      commentText: [],
+      logedInUser: localStorage.getItem('user'),
+    }
+  }
+
+  logOut = () => {
+    localStorage.setItem('user', 'nope')
+    this.props.logOutTop()
+    this.setState({
+      logedInUser : ''
+    });
+  }
+  componentDidMount(){
+    this.setState({
+     data: dummyData 
+    })
+  };
+
+  handleInputChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+      });
+
+  };
+
+  handleInputChangeComment = event => {
+    let copy = this.state.commentText.slice()
+    copy[event.target.name] = event.target.value
+    
+    this.setState({
+      commentText : copy
+    });
+  };
+
+  addNewComment = (event, index) => {
+    event.preventDefault();
+    
+    let newComment = { text: this.state.commentText[index], username: this.state.logedInUser};
+    let newDataState = this.state.data;
+    newDataState[index].comments.push(newComment);
+    let newCommentArray = this.state.commentText;
+    newCommentArray[index] = ''
+
+    this.setState({
+      data: newDataState,
+      commentText: newCommentArray
+    });
+  };
+
+  removeComment = (event, parentIndex, childIndex) => {
+    event.preventDefault();
+
+    let newDataState = this.state.data;
+
+    newDataState[parentIndex].comments.splice(childIndex, 1);
+
+    this.setState({
+      data: newDataState,
+    });
+  }
+
+  addLike = (event, index) => {
+    event.preventDefault();
+
+    let newDataState = this.state.data;
```

Comment:
You're not actually creating a copy of the state data here you're just creating a new reference.  In other words this is directly mutating your state.